server/src/test/java/org/opensearch/gateway/remote/RemoteClusterStateServiceTests.java

```diff
@@ -658,6 +737,991 @@ public void testWriteIncrementalMetadataSuccessWhenPublicationEnabled() throws I
         assertThat(manifest.getIndicesRouting().size(), is(1));
     }
 
+    public void testTimeoutWhileWritingMetadata() throws IOException {
+        AsyncMultiStreamBlobContainer container = (AsyncMultiStreamBlobContainer) mockBlobStoreObjects(AsyncMultiStreamBlobContainer.class);
+        doNothing().when(container).asyncBlobUpload(any(), any());
+        int writeTimeout = 2;
+        Settings newSettings = Settings.builder()
+            .put("cluster.remote_store.state.global_metadata.upload_timeout", writeTimeout + "s")
```

Comment:
How do we know that the upload will not complete within 2s ?

Reply:
The latched listener will not receive any response or failure and we still be waiting for latch countdown. So, reducing the time we will wait to reduce the test time.